packages/devp2p/test/integration/rlpx-simulator.spec.ts

```diff
@@ -6,76 +6,63 @@ import { DISCONNECT_REASON } from '../../src/types.js'
-        udpPort: 30308,
-        tcpPort: null,
-      }
-      assert.notOk(
-        rlpxs[0]['_dpt']!['_banlist'].has(peer),
-        'should not be in ban list before bad peer discovered',
-      )
-      rlpxs[0]['_dpt']!.events.emit('peer:new', peer)
-      assert.ok(
-        rlpxs[0]['_dpt']!['_banlist'].has(peer),
-        'should be in ban list after bad peer discovered',
-      )
-      await util.delay(500)
-      util.destroyRLPXs(rlpxs)
+    await new Promise((resolve) => {
+      rlpxs[0].events.on('peer:added', async (peer) => {
+        assert.equal(
+          peer['_port'],
+          basePort + 1,
+          'should have added peer on peer:added after successful handshake',
+        )
+        assert.equal(rlpxs[0].getPeers().length, 1, 'peer list length should be 1')
+        assert.equal(rlpxs[0]._getOpenSlots(), 9, 'should have maxPeers - 1 open slots left')
+        await util.delay(500)
+        util.destroyRLPXs(rlpxs)
+        resolve(undefined)
+      })
     })
   })
-
-  it('RLPX: remove node', () => {
-    const rlpxs = util.initTwoPeerRLPXSetup(undefined, undefined, undefined, 40504)
-
-    try {
-      rlpxs[0].events.once('peer:added', (peer) => {
-        rlpxs[0].disconnect(peer['_remoteId'])
-      })
-      rlpxs[0].events.once('peer:removed', async (_, reason: any) => {
-        assert.equal(
-          reason,
-          DISCONNECT_REASON.CLIENT_QUITTING,
-          'should close with CLIENT_QUITTING disconnect reason',
+  it('RLPX: ban node with missing tcp port', async () => {
+    const { rlpxs, peer } = util.initTwoPeerRLPXSetup(undefined, undefined, undefined, 40444)
+    rlpxs[0]['_dpt']!.addPeer(peer).catch(() => {
+      throw new Error('Peering failed')
+    })
+    await new Promise((resolve) => {
+      rlpxs[0].events.on('peer:added', async () => {
+        const peer = {
+          id: hexToBytes('0xabcd'),
+          address: '127.0.0.1',
+          udpPort: 30308,
+          tcpPort: null,
+        }
+        assert.notOk(
+          rlpxs[0]['_dpt']!['_banlist'].has(peer),
+          'should not be in ban list before bad peer discovered',
+        )
+        rlpxs[0]['_dpt']!.events.emit('peer:new', peer)
+        assert.ok(
+          rlpxs[0]['_dpt']!['_banlist'].has(peer),
+          'should be in ban list after bad peer discovered',
         )
-        assert.equal(rlpxs[0]._getOpenSlots(), 10, 'should have maxPeers open slots left')
         await util.delay(500)
         util.destroyRLPXs(rlpxs)
+        resolve(undefined)
       })
-    } catch (err) {
-      assert.fail(`An unexpected error occurred: ${err}`)
-    }
+    })
   })
-
-  it('RLPX: test peer queue / refill connections', () => {
+  it('RLPX: test peer queue / refill connections', async () => {
     const basePort = 60661
     const rlpxs = util.getTestRLPXs(3, 1, basePort)
     const peer = { address: util.localhost, udpPort: basePort + 1, tcpPort: basePort + 1 }
     rlpxs[0]['_dpt']!.addPeer(peer)
-    try {
```

Comment:
I reworked this test case to be a lot more similar to how the test was originally written to preserve the test coverage it gives for queuing/refilling connections while also updating it to run asynchronously.